plugin-filters/finaid-post-list-layout.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * Provides post list layouts for grouped
+ */
+namespace FinAid\Utils\Plugins;
+
+/**
+ * Function for displaying the beginning of a grouped post list
+ * @author Jim Barnes
+ * @since 1.0.0
+ * @param string $content The content passed to the filter
+ * @param array $posts The posts
+ * @param array $atts The shortcode arguments
+ * @return string
+ */
+function ucf_post_list_display_grouped_before( $content, $posts, $atts ) {
+	ob_start();
+?>
+	<div class="ucf-post-list ucf-post-list-grouped" id="post-list-<?php echo $atts['list_id']; ?>">
+<?php
+	return ob_get_clean();
+}
+
+add_filter( 'ucf_post_list_display_grouped_before', array( __NAMESPACE__ . "\ucf_post_list_display_grouped_before" ), 10, 3 );
+
+/**
+ * Function for displaying the title of the grouped post list
+ * @author Jim Barnes
+ * @since 1.0.0
+ * @param string $content The content passed to the filter
+ * @param array $posts The posts
+ * @param array $atts The shortcode arguments
+ * @return string
+ */
+function ucf_post_list_display_grouped_title( $content, $posts, $atts ) {
+	$formatted_title = '';
+
+	if ( $list_title = $atts['list_title'] ) {
+		$formatted_title = "<h2 class=\"ucf-post-list-title\">$list_title</h2>";
+	}
+
+	return $formatted_title;
+}
+
+add_filter( 'ucf_post_list_display_grouped_title', __NAMESPACE__ . '\ucf_post_list_display_grouped_title', 10, 3 );
+
+/**
+ * Function for displaying the items of the grouped post list
+ * @author Jim Barnes
+ * @since 1.0.0
+ * @param string $content The content passed to the filter
+ * @param array $posts The posts
+ * @param array $atts The shortcode arguments
+ * @return string
+ */
+function ucf_post_list_display_grouped( $content, $posts, $atts ) {
+	$group_by_subterm = false;
+
+	if ( isset( $atts['group_by_subterm'] ) ) {
+		$group_by_subterm = true;
+	}
+
+	// If we're grouping by subterm, go ahead and do it
+	list($grouped, $posts) = $group_by_subterm ? ucf_post_list_grouped_groupby_subterm( $posts, $atts ) : array(false, $posts);
+
+	ob_start();
+
+	if ( $grouped ) :
+		foreach( $posts as $term ) :
+?>
+	<h2 class="heading-underline"><?php echo $term['term_name']; ?></h2>
+	<ul class="mb-2 list-unstyled">
```

Comment:
What do you think about increasing this margin-bottom to `mb-4`? The extra space helps separate out the groupings visually.

Reply:
Updated!